### PR TITLE
Rm react test renderer 3

### LINKED ISSUE #774: Remove use of react-test-renderer
We currently have 2 test renderers in Psammead and Simorgh - `react-test-renderer` and `@testing-library/react`.

Making use of 2 test renderers will lead to confusion about which one to use and when, especially when they have different ideas about how to write tests.

## Solution
Remove `react-test-renderer` and all use of it. Refactor shallow render tests because `@testing-library/react` does not have or encourage shallow rendering.

---

Some of the benefits of using `@testing-library/react` include:
 - encourages better testing practices and writing more maintainable tests 
 - is designed to enable and encourage writing tests that use components as the end users use them
 - it supports querying for elements in ways that encourage accessibility in components

Find out more about [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro).

This article briefly describes and compares React test renders https://codeburst.io/revisiting-react-testing-in-2019-ee72bb5346f4

Here is more on why [`@testing-library/react` doesn't do shallow rendering](https://kentcdodds.com/blog/why-i-never-use-shallow-rendering).

### PR DESCRIPTION
**Pull request by mykeels**
Resolves https://github.com/bbc/psammead/issues/774

Duplicate of #4269

**Overall change:** _Remove use of `react-test-renderer`, and use `@testing-library/react` instead._

**Code changes:**

- _Remove use of `react-test-renderer`, and use `@testing-library/react` instead._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/ChartbeatAnalytics/index.test.jsx [hunk 1]
```diff
@@ -120,35 +118,31 @@ describe('Charbeats Analytics Container', () => {
       {},
     );
     expect(testUtils.getConfig).toHaveBeenCalledTimes(1);
-    expect(tree).toMatchSnapshot();
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (jroebu14) [inline src/app/containers/ChartbeatAnalytics/index.test.jsx:121]: can we not use the shouldMatchSnapshot helper from pasammead-test-helpers?

File: src/app/containers/ChartbeatAnalytics/canonical/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -1,1543 +1,13 @@
...
-                >
-                  
-        (function(){
-          var _sf_async_config = window._sf_async_config = (window._sf_async_config || {});
-          var config = {"domain":"test-domain","type":"article","sections":"section1 section2","chartbeatUID":1111,"virtualReferrer":null,"useCanonical":true,"title":"This is an article","uid":123};
-          for (var key in config) {
-            _sf_async_config[key] = config[key];
-          }
-        })();
+<html>
```
Comment (jroebu14) [inline src/app/containers/ChartbeatAnalytics/canonical/__snapshots__/index.test.jsx.snap:4]: this is so much better than the old snapshot. I can't believe the old snapshot was even added!

**Review by EinsteinNjoroge — COMMENTED** [review]
Love this approach.

File: src/app/containers/ChartbeatAnalytics/index.test.jsx (hunk 1 above)
Comment (mykeels) [reply src/app/containers/ChartbeatAnalytics/index.test.jsx:121]: There are other asserts in the test, so I'd rather make that change in another PR if possible

**Comment by mykeels** [comment]
Closing because this branch contains an unverified commit.

**Pull request closed without merging**